test/unit/test_system_client.py

```diff
@@ -185,3 +188,55 @@ def tearDown(self):
             shutil.rmtree(self.base_dir)
         except:
             pass
+
+
+class TestUntarPackage(unittest.TestCase):
```

Comment:
Love the strategies you've employed here. Making the tarballs in the tests is much cleaner than my previous suggestion of committing tarball files.